### DISCUSSION TITLE
add Mailpile as webmail replacement

### DISCUSSION THREAD

**Discussion by hadifarnoud** [modoboa/modoboa#2134] [Ideas]
can we add Mailpile as a plugin to installer? what is needed to be done? it has added multi-user support and this will make Modoboa a robust mail service.

**Comment by tonioo** [comment]
Yes it is possible. Any pull request about this would be appreciated ;)

**Comment by hadifarnoud** [comment]
I just need some help on how to do it. there is no docs for adding plugin to installer

**Comment by almereyda** [comment]
As far as I am [watching the Mailpile development](https://github.com/mailpile/Mailpile/graphs/contributors) in the last three years, it happens to be a little stuck lately.
Also it needs a separate instance per user, due to its GPG inclusion. This would need special orchestration for each individual user and is originally supposed to run locally.
I suggest rather inspecting other common webmail offers like Rouncube or Squirrelmail. From a UI perspective, Rainloop may also be interesting.

But just not Mailpile, as far as I am concerned.

**Comment by hadifarnoud** [comment]
@almereyda either option is good. Mailpile [added](https://github.com/mailpile/Mailpile/issues/820) multi-user functionality recently. I did not test it myself tbh.

I saw Rainloop too, it's great. the issue is I need an example on how to add another webmail client as plugin. I think others can use that to add any client they like.

**Comment by almereyda** [comment]
As we also move to a self-hosted webmail component, which attaches to the services via SMTPS and IMAPS (https://github.com/ecobytes/docker-roundcube), I see this issue evolving into the direction of how to add webmail replacements, instead of already specifying specific ones.

Due to this, @gandhiano and me have an interest to turn the webmail tab into a simple link to, i.e., https://webmail.ecobytes.net

Should we add such an option to a configuration pane of https://github.com/modoboa/modoboa-webmail/, and create an issue for that in https://github.com/modoboa/modoboa-webmail/issues ?

**Comment by tonioo** [comment]
@almereyda Installing the webmail component only to bypass it looks pretty strange to me... What do you think about the possibility to add custom links into the top menu?

**Comment by hadifarnoud** [comment]
@tonioo it's a good idea to have it installed so people can go back to it if for some reason other client stops working. having custom menu would also be very useful imho
